src/schema/v2/userInquiredArtworks.ts

```diff
@@ -0,0 +1,47 @@
+import { connectionWithCursorInfo } from "schema/v2/fields/pagination"
+import { ResolverContext } from "types/graphql"
+import {
+  GraphQLString,
+  Thunk,
+  GraphQLFieldConfigMap,
+  GraphQLBoolean,
+} from "graphql"
+import { IDFields } from "./object_identification"
+import { ArtworkType } from "./artwork"
+import { date } from "./fields/date"
+
+export const edgeFields: Thunk<GraphQLFieldConfigMap<
+  any,
+  ResolverContext
+>> = () => ({
+  ...IDFields,
+  status: {
+    type: GraphQLString,
+    description:
+      "This reflects the `title` attribute of the most recent embedded object in `statuses`",
+    resolve: ({ statuses }) => {
+      const statusesWithDates = statuses.map((obj) => {
+        return { ...obj, createdAt: new Date(obj.created_at) }
+      })
+
+      const sortedStatusesReverseChronological = [...statusesWithDates].sort(
+        (objA, objB) => objB.createdAt.getTime() - objA.createdAt.getTime()
+      )
+
+      return sortedStatusesReverseChronological[0]?.title
```

Comment:
Just want to make sure I'm understanding the modeling. The statuses will be an array of objects, where each object is an `ArtworkInquiryRequest` in gravity (the node will be the Artwork inquired upon), correct? The sorting logic is to just return the most recent status for a specific inquiry.

Reply:
Almost, each object in the arrray of statuses, is an instance of `ArtworkInquiryRequestStatus` (note the status suffix) which is an embedded model inside an inquiry - https://github.com/artsy/gravity/blob/f40e4f08d01b688b6f0ca2c47dd6f875efcca45d/app/models/domain/inquiry_request_status.rb#L5

The `statuses` argument represents the `statuses` attribute from the JSON of an inquiry request, which is the array of statuses: https://github.com/artsy/gravity/blob/f40e4f08d01b688b6f0ca2c47dd6f875efcca45d/app/models/domain/inquiry_request.rb#L372 (but it seems like the thing we care about is the most recent status, and in particular the `title`).

It's very interesting modeling! 😆

Reply:
Oh and also! This concept ('latest status') was ported from Torque: https://github.com/artsy/torque/blob/23afe1804740f0720799eca5f3fe9c4d68f00b4f/app/client/models/artwork_inquiry_request.coffee#L60-L61 , the `title` property from that was what was being rendered on the user page.